src/Main.hs

```diff
@@ -1,29 +1,57 @@
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
 
 module Main where
 
 import Control.Concurrent.MVar (takeMVar, newMVar)
-import Data.Maybe              (listToMaybe)
 import Data.MultiSet           (distinctSize)
-import Data.Text               (pack)
-import System.Environment      (getArgs)
+import Data.Text               (Text, pack)
+import Data.Semigroup          ((<>))
 
 import Echidna.Exec
 import Echidna.Solidity
 
 import Hedgehog hiding (checkParallel)
 import Hedgehog.Internal.Property (GroupName(..), PropertyName(..))
 
+import Options.Applicative
+
+data Options = Options
+  { filePath         :: FilePath
+  , selectedContract :: Maybe String
+  , solcArgs         :: Maybe String }
+
+options :: Parser Options
+options = Options
+      <$> argument str
+          ( metavar "FILE"
+         <> help "Solidity file to analyze" )
+      <*> optional ( argument str
+          ( metavar "CONTRACT"
+         <> help "Contract inside of file to analyze" ))
+      <*> optional ( strOption
+          ( long "solc-args"
+         <> help "Optional solidity compiler arguments" ))
+
+maybePack :: Maybe String -> Maybe Text
```

Comment:
I think this is just `fmap pack`/`pack <$>`, we probably don't need to define it like this